daft/table/table.py

```diff
@@ -114,14 +114,39 @@ def concat(cls, to_merge: list[Table]) -> Table:
     ###
 
     def to_arrow(self) -> pa.Table:
-        python_fields = [field for field in self.schema() if field.dtype == DataType.python()]
+        python_fields = {field.name for field in self.schema() if field.dtype == DataType.python()}
         if python_fields:
-            raise ValueError(f"Cannot convert table to Arrow due to presence of Python columns f{python_fields}")
-        return pa.Table.from_batches([self._table.to_arrow_record_batch()])
+            table = {}
+            for colname in self.column_names():
+                column_series = self.get_column(colname)
+                if colname in python_fields:
+                    # TODO(Clark): Get the column as a top-level ndarray to ensure it remains contiguous.
+                    column = column_series.to_pylist()
```

Comment:
If this Python object column _isn't_ a tensor column, we will still do a best-effort conversion to an Arrow table via the pyarrow-level type inference done in `pa.Table.from_pydict()`. If that fails, that error will be surfaced to the user.

We could add more defensive checked here to provide a better error message to the user, but this seems fine to me as a first go.